test/integration/model/find.test.js

```diff
@@ -760,104 +759,71 @@ describe(Support.getTestDialectTeaser('Model'), function() {
-                , tags = results.tags;
-
-              async.parallel([
-                function(callback) {
-                  products[0].setTags([tags[0], tags[1]]).done(callback);
-                },
-                function(callback) {
-                  products[1].addTag(tags[0]).done(callback);
-                },
-                function(callback) {
-                  products[2].addTag(tags[1]).done(callback);
-                },
-                function(callback) {
-                  products[3].setTags([tags[1]]).done(callback);
-                },
-                function(callback) {
-                  products[4].setTags([tags[2]]).done(callback);
-                }
-              ], function(err) {
-                expect(err).not.to.exist;
-
-                async.parallel([
-                  function(callback) {
-                    self.Tag.find({
-                      where: {
-                        id: tags[0].id
-                      },
-                      include: [
-                        {model: self.Product, as: 'products'}
-                      ]
-                    }).done(function(err, tag) {
-                      expect(tag).to.exist;
-                      expect(tag.products.length).to.equal(2);
-                      callback();
-                    });
-                  },
-                  function(callback) {
-                    tags[1].getProducts().done(function(err, products) {
-                      expect(products.length).to.equal(3);
-                      callback();
-                    });
-                  },
-                  function(callback) {
-                    self.Product.find({
-                      where: {
-                        id: products[0].id
-                      },
-                      include: [
-                        {model: self.Tag, as: 'tags'}
-                      ]
-                    }).done(function(err, product) {
-                      expect(product).to.exist;
-                      expect(product.tags.length).to.equal(2);
-                      callback();
-                    });
-                  },
-                  function(callback) {
-                    products[1].getTags().done(function(err, tags) {
-                      expect(tags.length).to.equal(1);
-                      callback();
-                    });
-                  }
-                ], done);
+          return this.sequelize.sync().then(function() {
+            return Promise.all([
+              self.Product.bulkCreate([
+                {title: 'Chair'},
+                {title: 'Desk'},
+                {title: 'Handbag'},
+                {title: 'Dress'},
+                {title: 'Jan'}
+              ]),
+              self.Tag.bulkCreate([
+                {name: 'Furniture'},
+                {name: 'Clothing'},
+                {name: 'People'}
+              ])
+            ]).then(function() {
+              return Promise.all([
```

Comment:
The way it worked ealier with async.auto was that it created the entries and ran a .findAll and returned that as a result property. I separated that by first creating all entries and then searching for the different created values.
So it's about the second promise.all and likely the best to handle it would be to change the .all to .join, since that's the fasted way and the result is not needed at all. Only the propertys are used, as in either:
```javascript
Promise.all([first, second]).spread(function(first, second) {});
```
or
```javascript
Promise.join(first, second, function(first, second) {});
```
And props would look like this (with a bunch of code you do not need in my opinion):
```
Promise.props({
    first: first,
    second: second
}).then(function(result) {
    var first = result.first;
    var second = result.second;
});
```